packages/astro/src/core/build/plugins/plugin-pages.ts

```diff
@@ -6,12 +6,12 @@ import { eachPageData, hasPrerenderedPages, type BuildInternals } from '../inter
 import type { AstroBuildPlugin } from '../plugin';
 import type { StaticBuildOptions } from '../types';
 
-export function vitePluginPages(opts: StaticBuildOptions, internals: BuildInternals): VitePlugin {
+function vitePluginPages(opts: StaticBuildOptions, internals: BuildInternals): VitePlugin {
 	return {
 		name: '@astro/plugin-build-pages',
 
 		options(options) {
-			if (opts.settings.config.output === 'static' || hasPrerenderedPages(internals)) {
```

Comment:
`vitePluginPage` runs before `vitePluginPrerender` which is responsible for parsing the `prerender` metadata of astro files, so there's no way to know if there are prerendered pages or not.
This would always return `undefined` so I removed it.